### ISSUE TITLE
No ability to override region via CLI/env

### ISSUE THREAD

**Issue by philpennock** [99designs/aws-vault#237]
As opposed to #156 I have a setup where I have a default region in `~/.aws/config` and everything mostly works great.  (Really love `aws-vault`, it's awesome, thank you!).  Version: v4.2.0   Platform: macOS

Right up until I want to use another region with a given profile, to which I've already authenticated.  And run right into a roadblock.  Without `aws-vault` in the mix, I can just use `AWS_DEFAULT_REGION=xx-yyyy-n my-aws-clienttool` and the region from the environment will override the region from the config file.  With `aws-vault`, the precedence is inverted because the config value becomes the environment value, stomping on whatever was there before, which means there's no easy way to use a given auth profile but with a different region.

I can duplicate the profile with a different region in `~/.aws/config`, which works, but requires another round of MFA authentication and is ugly when it comes to setups where there are more than two regions in use.  I'd prefer to avoid this.

Ideally, `aws-vault` would examine environment and if it sees `AWS_DEFAULT_REGION` _already_ in environment, then it would honor that instead of the config file, to preserve the existing precedence rules.

Less ideally, having a flag so `aws-vault exec --region xx-yyyy-n cmd param1` works would work for me.  I have an `aws` wrapper script to invoke `aws-vault` automatically, so I can put any auxilliary logic there, to munge the existing profile into a parameter.

I can write a patch if that's preferred, just let me know which of the above approaches you'd find acceptable for inclusion.

Thanks.

**Comment by virgilwashere** [comment]
Yes please.  Use environment to override `~/.aws/config`

**Comment by davidneudorfer** [comment]
@philpennock Can you prepend `aws-vault` with `AWS_REGION=xx-yyyy-n`?

**Comment by philpennock** [comment]
I couldn't when I wrote the report, I haven't checked recently.  I tried both `AWS_REGION` and `AWS_DEFAULT_REGION` and `aws-vault` stomped on one and the other was just ignored.  I think.  I'm going from memory here.  I don't currently have anything up in AWS to check against.

But yes, prepending the command-line with `FOO=bar` is the bash/zsh way of setting an environment variable for that one command invocation, and the point of this Issue is that `aws-vault` is stomping over that.
